and-cli-deploy-azure-web-app.js

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env node
+require("./command-runner").run(async () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Imports
+    // -----------------------------------------------------------------------------------------
+
+    const azure         = require("./_modules/azure");
+    const dotnetPath    = require("./_modules/dotnet-path");
+    const echo          = require("./_modules/echo");
+    const program       = require("commander");
+    const shell         = require("shelljs");
+
+    // #endregion Imports
+
+    // -----------------------------------------------------------------------------------------
+    // #region Variables
+    // -----------------------------------------------------------------------------------------
+
+    let   appName             = null;
+    let   branch              = null;
+    let   clientId            = null;
+    let   force               = false;
+    const pythonInstallerUrl  = "https://www.python.org/ftp/python/3.7.4/python-3.7.4-amd64.exe";
+    let   remote              = null;
+    let   resourceGroup       = null;
+    let   secret              = null;
+    let   tenantId            = null;
+    let   username            = null;
+
+    // #endregion Variables
+
+    /// -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+
+    const deployAzureWebApp = {
+        createRemoteIfMissing() {
+            if (shell.exec(`git remote get-url ${remote}`).code !== 0) {
```

Comment:
Suggestion: Recommend extracting this exec(...).code into local variable and then short-circuiting in an effort to reduce nesting